streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -877,23 +881,46 @@ private void connectProcessorAndStateStore(String processorName, String stateSto
         return sourceTopics;
     }
 
-    private void connectStateStoreNameToSourceTopics(final String stateStoreName,
-                                                     final ProcessorNodeFactory processorNodeFactory) {
+    private Set<Pattern> findSourcePatternsForProcessorParents(String[] parents) {
```

Comment:
I'm wondering if we can not add this method and slightly modify the one above it that is just about identical? i.e., this is just a thought but i think it could look something like this:
```
interface SourceCollector {
        void collect(final SourceNodeFactory factory);
    }

private void findSourcesForProcessorParents(String[] parents, SourceCollector sourceCollector) {
        for (String parent : parents) {
            NodeFactory nodeFactory = nodeFactories.get(parent);
            if (nodeFactory instanceof SourceNodeFactory) {
                sourceCollector.collect((SourceNodeFactory) nodeFactory);
            } else if (nodeFactory instanceof ProcessorNodeFactory) {
                findSourcePatternsForProcessorParents(((ProcessorNodeFactory) nodeFactory).parents, sourceCollector);
            }
        }
    }
```